pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -107,74 +106,62 @@ func TestDevicePluginReRegistration(t *testing.T) {
 		{ID: "Dev3", Health: pluginapi.Healthy},
 	}
 	for _, preStartContainerFlag := range []bool{false, true} {
-
-		expCallbackCount := int32(0)
-		callbackCount := int32(0)
-		callbackChan := make(chan int32)
-		callback := func(n string, a, u, r []pluginapi.Device) {
-			callbackCount++
-			if callbackCount > atomic.LoadInt32(&expCallbackCount) {
-				t.FailNow()
-			}
-			callbackChan <- callbackCount
-		}
-		m, p1 := setup(t, devs, callback, socketName, pluginSocketName)
-		atomic.StoreInt32(&expCallbackCount, 1)
+		m, ch, p1 := setup(t, devs, nil, socketName, pluginSocketName)
 		p1.Register(socketName, testResourceName, preStartContainerFlag)
-		// Wait for the first callback to be issued.
 
 		select {
-		case <-callbackChan:
-			break
-		case <-time.After(time.Second):
-			t.FailNow()
+		case <-ch:
+		case <-time.After(5 * time.Second):
+			t.Fatalf("timeout wait for manager update")
 		}
-		devices := m.Devices()
-		require.Equal(t, 2, len(devices[testResourceName]), "Devices are not updated.")
+
+		_, allocatable, _ := m.GetCapacity()
+		resourceAllocatable, _ := allocatable[v1.ResourceName(testResourceName)]
+		require.Equal(t, int64(2), resourceAllocatable.Value(), "Devices are not updated.")
 
 		p2 := NewDevicePluginStub(devs, pluginSocketName+".new")
 		err = p2.Start()
 		require.NoError(t, err)
-		atomic.StoreInt32(&expCallbackCount, 2)
 		p2.Register(socketName, testResourceName, preStartContainerFlag)
-		// Wait for the second callback to be issued.
+
 		select {
-		case <-callbackChan:
-			break
-		case <-time.After(time.Second):
-			t.FailNow()
+		case <-ch:
+		case <-time.After(5 * time.Second):
+			t.Fatalf("timeout wait for manager update")
 		}
 
-		devices2 := m.Devices()
-		require.Equal(t, 2, len(devices2[testResourceName]), "Devices shouldn't change.")
+		_, allocatable, _ = m.GetCapacity()
+		resourceAllocatable, _ = allocatable[v1.ResourceName(testResourceName)]
+		require.Equal(t, int64(2), resourceAllocatable.Value(), "Devices shouldn't change.")
```

Comment:
will it add value if we check capacity as well?

Reply:
done, thanks